src/inspect.jl

```diff
@@ -26,8 +26,16 @@ function AbstractTrees.nodevalue(x::BasicSymbolic)
     Text(str)
 end
 
+"""
+$(TYPEDSIGNATURES)
+
+Return the children of the symbolic expression `x`, sorted by their order in
+the expression.
+
+This function is used internally for printing via AbstractTrees.
+"""
 function AbstractTrees.children(x::Symbolic)
-    iscall(x) ? arguments(x) : isexpr(x) ? children(x) : ()
+    iscall(x) ? arguments(x; sort = true) : isexpr(x) ? children(x) : ()
```

Comment:
see https://github.com/JuliaSymbolics/TermInterface.jl/pull/44#issuecomment-2178329326

Reply:
Shouldn't this not force sort then, given that would be `sorted_children`?

Reply:
Which option do you prefer?
1. Adding a `sort` keyword argument.
2. Creating new functions `sorted_arguments` and `sorted_children`.

Also, the `arguments` and `children` functions in SymbolicUtils.jl v2.0.2 aren't related to TermInterface.jl yet.

Reply:
Let's get the TermInterface.jl change in and hit that interface. Since that's happening like tomorrow there's no need to do a separate version as well.

Reply:
> Which option do you prefer?
> 
>     1. Adding a `sort` keyword argument.
> 
>     2. Creating new functions `sorted_arguments` and `sorted_children`.
> 
> 
> Also, the `arguments` and `children` functions in SymbolicUtils.jl v2.0.2 aren't related to TermInterface.jl yet.

@bowenszhu TermInterface.jl 2.0 was merged with `sorted_argument` and `sorted_children`.  I have a PR open for TermInterface.jl 1.0, which should be updated to 2.0 - want update this MR to target that branch? I'll be at a conference and won't have much time.